### PR TITLE
[MB-1706] Fix for Expiry Message Deletion when there is no subscription

### PR DESCRIPTION
**Pull request by sirojan24**
Associated Jira: https://wso2.org/jira/browse/MB-1706

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesContextStore.java [hunk 1]
```diff
@@ -435,6 +435,13 @@ void setSlotState(long startMessageId, long endMessageId, SlotState slotState)
     Set<String> getAllQueues() throws AndesException;
 
     /**
+     * Get all queue names for which messages are published
+     * @return
```
Comment (sdkottegoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesContextStore.java]: Shall we have correct formatting? And also, add what's returned

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/coordination/SlotAgent.java [hunk 2]
```diff
@@ -231,6 +231,13 @@ void setSlotState(long startMessageId, long endMessageId, SlotState slotState)
 	Set<String> getAllQueues() throws AndesException;
 
 	/**
+	 * Get list of all queues in submitted slots
+	 * @return Set of queue names
```
Comment (sdkottegoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/coordination/SlotAgent.java]: There needs to be a new line in between

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSAndesContextStoreImpl.java [hunk 3]
```diff
@@ -1867,6 +1867,39 @@ public void deleteMessageId(long messageId) throws AndesException {
...
+    @Override
+    public Set<String> getAllQueuesInSubmittedSlots() throws AndesException {
+        Connection connection = null;
+        PreparedStatement preparedStatement = null;
+        ResultSet resultSet = null;
+        Set<String> queueList = new TreeSet<>();
+
+        try {
+            connection = getConnection();
+
```
Comment (sdkottegoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSAndesContextStoreImpl.java]: Unnecessary new line

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSAndesContextStoreImpl.java [hunk 4]
```diff
@@ -1867,6 +1867,39 @@ public void deleteMessageId(long messageId) throws AndesException {
...
+            resultSet = preparedStatement.executeQuery();
+
+            while (resultSet.next()) {
+                queueList.add(resultSet.getString(RDBMSConstants.QUEUE_NAME));
+            }
+            return queueList;
+
+        } catch (SQLException e) {
+            String errMsg =
+                    RDBMSConstants.TASK_GET_ALL_QUEUES_IN_SUBMITTED_SLOTS;
```
Comment (sdkottegoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSAndesContextStoreImpl.java]: Can be merged into a single line

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSConstants.java [hunk 5]
```diff
@@ -731,6 +731,13 @@
             + " FROM " + SLOT_TABLE;
 
     /**
+     * Prepared statement to get all queues in submitted slots
+     */
+    protected static final String PS_GET_ALL_QUEUES_IN_SUBMITTED_SLOTS =
+            "SELECT DISTINCT " + QUEUE_NAME
```
Comment (sdkottegoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSConstants.java]: Let's use the alignement followed in the other statements

Comment (sirojan24) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSConstants.java]: Done :)

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSAndesContextStoreImpl.java [hunk 6]
```diff
@@ -1867,6 +1867,36 @@ public void deleteMessageId(long messageId) throws AndesException {
...
+     */
+    @Override
+    public Set<String> getAllQueuesInSubmittedSlots() throws AndesException {
+        Connection connection = null;
+        PreparedStatement preparedStatement = null;
+        ResultSet resultSet = null;
+        Set<String> queueList = new TreeSet<>();
+        try {
+            connection = getConnection();
+
```
Comment (sdkottegoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSAndesContextStoreImpl.java]: Unnecessary new line

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesContextStore.java [hunk 7]
```diff
@@ -435,6 +435,13 @@ void setSlotState(long startMessageId, long endMessageId, SlotState slotState)
     Set<String> getAllQueues() throws AndesException;
 
     /**
+     * Get all queue names for which messages are published
```
Comment (sirojan24) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesContextStore.java]: Fixed in https://github.com/wso2/andes/pull/714

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSAndesContextStoreImpl.java (hunk 6 above)
Comment (sirojan24) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSAndesContextStoreImpl.java]: Fixed in https://github.com/wso2/andes/pull/714

**Pull request merged**